### PR TITLE
#15 Enable SSL certificate checks

### PR DESCRIPTION
**Pull request by unrealwork**
#15 Now in case of SSL errors, the message will be displayed in the preview section

image

### CODE REVIEW

File: client/resources/js/config_init.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Initialize portal with config from
+ */
+if (typeof initializePortal === "function") {
+    console.log(previewOptions);
```
Comment (kirmanak) [inline client/resources/js/config_init.js]: I guess, you've forgotten to remove the `console.log` call

**Comment by rodionos** [comment]
The screenshot looks good.

File: client/src/axibaseChartsProvider.ts [hunk 2]
```diff
@@ -129,73 +129,37 @@ ${this.text.substr(match.index + match[0].length + 1)}`;
             .get("colorTheme");
         let darkTheme: string = "";
         if (theme && /[Bb]lack|[Dd]ark|[Nn]ight/.test(theme)) {
-            darkTheme = `<link rel="stylesheet" type="text/css"
```
Comment (kirmanak) [inline client/src/axibaseChartsProvider.ts]: Why did you remove `type="text/css"`?

File: client/src/axibaseChartsProvider.ts [hunk 3]
```diff
@@ -129,73 +129,37 @@ ${this.text.substr(match.index + match[0].length + 1)}`;
...
 	<script>
-		if (typeof initializePortal === "function") {
-			initializePortal(function (callback) {
-				var configText = ${JSON.stringify(this.text)};
-				if (typeof callback === "function") {
-					callback([configText, portalPlaceholders = getPortalPlaceholders()]);
-				}
-			});
-		}
+	    window.previewOptions = ${JSON.stringify({
```
Comment (kirmanak) [inline client/src/axibaseChartsProvider.ts]: Why do you use `JSON.stringify()` here? In my opinion, you should `stringify` only `this.text` [here](https://github.com/axibase/axibase-charts-vscode/blob/b6cdb1964de5fce28e85564bd30a5e19c3251ab1/client/resources/js/config_init.js#L7)

File: client/resources/js/init.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ */
+window.initChart = function () {
+    const pingUrl = `${previewOptions.url}/api/v1/ping${previewOptions.jsessionid ?
+        `;jsessionid=${previewOptions.jsessionid}` : ""}`;
+    $.get(pingUrl, () => {
+        if (onBodyLoad) {
+            onBodyLoad();
+        } else {
+            const $body = $('body');
+            $body.empty().append('<h3>Corrupted charts.js files</h3>');
```
Comment (kirmanak) [inline client/resources/js/init.js]: `charts.min.js`

File: client/src/axibaseChartsProvider.ts (hunk 2 above)
Comment (unrealwork) [reply client/src/axibaseChartsProvider.ts]: It's not required by spec of HTML5 https://www.w3.org/TR/html40/struct/links.html#edef-LINK

File: client/src/axibaseChartsProvider.ts (hunk 3 above)
Comment (kirmanak) [reply client/src/axibaseChartsProvider.ts]: Okay, I've realized, this text in a string already, that's why you're using `stringify` here. But you have to stringify text additionally, don't you? What if `this.text` contains `"`?

Comment (unrealwork) [reply client/src/axibaseChartsProvider.ts]: It's just for passing all required args to window global variable, then I can use it in all my scripts.
It's just assigning the variable with JSON.

File: client/resources/js/init.js (hunk 4 above)
Comment (unrealwork) [reply client/resources/js/init.js]: I meant that one of our `charts` library files are incorrect.

File: client/src/axibaseChartsProvider.ts (hunk 3 above)
Comment (unrealwork) [reply client/src/axibaseChartsProvider.ts]: I'm sure that stringify escaped each property in JSON tree. Anyway I tried to do this, And it doesn't display for me in case of additional `stringify` if I've got you right.

```js
${JSON.stringify({
            jsessionid: this.jsessionid,
            text: JSON.stringify(this.text),
            url: this.url,
        })
```
